individual/models.py

```diff
@@ -13,12 +15,50 @@ class Individual(HistoryModel):
     #TODO WHY the HistoryModel json_ext was not enough
     json_ext = models.JSONField(db_column="Json_ext", blank=True, default=dict)
 
+    village = models.ForeignKey(
```

Comment:
I wonder if we should not have location instead, related names should be 'individuals'

Reply:
This is an interesting point. I thought about this long and hard – the pro is the flexibility, but the con is also the flexibility. Say we go with location and allow it to be defined at any of the 4 geo levels, on the UI for creating & updating an individual/group, we will have the user select the location type, followed by the name. For individual import, we'd have two columns: `location_type` and `location_name`, instead of one `village`. These are all ok.

Using location also allows individuals and groups to have different levels of granularity defined for them. Would this be good for data consistency from a record keeping point of view? e.g. one might end up with a database of 20k individuals' locations at district level while another 10k at village level – would this serve program administration well? @andreamartin @malike @amschel-de-r would be good to have your thoughts on this.

Another minor-ish point to consider – What would be the validation logic on the relationship between the individual.location and individual.group.location when they are both present? Should they match exactly? One contains another? Would there be enrollment implications if group and individual locations do not match exactly?

Reply:
yeah the concept of village as a geolocation is not available in all countries. The best approach will be to have labels which can be named at implementation level but tbh I don't know if its possible with the current openIMIS as the level of effort might be too big.

Reply:
@malike the term "village" can be customized through translation on the UI. The decision point here is to either pin it down to the lowest geo level out of the 4 (most specific) or leave it flexible to allow it to be defined at any geo level for each individual/group.

Reply:
I think it would be good practice that every instance has one level that is required for all individuals.

Having columns in data upload for `location_type` and `location_name` would add unnecessary confusion for all the countries I can think of. I would also be very surprised if there was a country that didn't have at least one geo level that all individuals have data on. 

What I would say, is that it might be good for the level the mandatory field is at (i.e. geo_4, geo_3, geo_5) to be customizable in one place, maybe in core_moduleConfigurations or something. This would help the system fit both a larger country - where they may have complete data down to geo_6 level and users that need row-level security at geo_3 level - as well as a small island country - where they may only have 3 geo levels.

If we're talking edge cases, e.g. migrants, refugees and IDPs, I think we can encourage a practice where they allow for a catch-all value. E.g. if the geo levels are Region, District, Municipality and Village, and refugees/IDPs are handled at the Region level, then for each region you would create e.g. R1D_NA - no district, R1M_NA - no municipality, R1V_NA no village. Then in the data upload, you could set all the refugees/migrants in district 1 to have R1V_NA and they would be successfully handled (if I understand the feature).

 _As a side note, having `village` as the mandatory column name would also confuse a lot - do we have somewhere where we can customize a mapping between mandatory model fields and csv upload column names?_

Open to comments - also have run this past Astrid as well